### ISSUE TITLE
CPUParticles2D, when instanced, sometimes just "sparkle" instead of emiting correctly

### ISSUE THREAD

**Issue by joaquin-montesdeoca** [godotengine/godot#25398]
**Godot version:**
v3.1.beta2.official

**OS/device including version:**
Windows 7 SP1 / GeForce GTX 1060 3GB (397.64)

**Issue description:**
I've created a scene with five CPUParticles2D nodes. When I instance it, sometimes some of the particles nodes won't emit particles properly. They would only "sparkle" a little as it's shown in the following images: [link](https://imgur.com/a/kXIdoo9).

**Steps to reproduce:**
Download the **Minimal reproduction project** and execute it. Left-click in the screen multiple times (this will instance the scene with the five CPUParticles2D nodes).  To clear the screen, right-click on it. 

This problem doesn't always happenm, so you'll probably have to click on the screen lots of times until the issue is visible.

**Minimal reproduction project:**
[cpu-particles-test-2.zip](https://github.com/godotengine/godot/files/2801203/cpu-particles-test-2.zip)

**Comment by clayjohn** [comment]
I am not seeing this behaviour in beta3. Can you download it and try again?

**Comment by joaquin-montesdeoca** [comment]
I'll check it out.

**Comment by joaquin-montesdeoca** [comment]
Dammit, I've updated to beta3 and I've gotten mixed results.

The **Minimal reproduction project** worked perfectly. No matter what I did, it worked fine. Which is great.

The thing is that, in the original game I detected the problem, most times it worked and sometimes the problem appeared again. Although a little diferent, the CPUParticles2D didn't "sparkle", they simply just didn't appear, as it's seen in the image: [link](https://imgur.com/a/BnygJRt).

I'll try to reproduce this behaviour again and if I get it right I'll report it again. Until then I'll close this ticket.

Thanks for your help.
